pytest_picked/modes.py

```diff
@@ -4,26 +4,78 @@
 
 
 class Mode(ABC):
-    def __init__(self, test_file_convention):
-        self.test_file_convention = test_file_convention
+    def __init__(self, test_conventions, only_modified_tests=False):
+        self.test_conventions = test_conventions
+        self.only_modified_tests = only_modified_tests
+
+    def only_tests(self, raw_output, re_list):
+        test_file_regex = r"(\/|^)" + r"|".join(re_list)
+        test_regex = ""
+        class_regex = ""
+        for item in self.test_conventions["function"]:
+            test_regex = (
+                test_regex + r"(?<=def\s).*{0}.*.[\w](?=\(|/:)".format(item) + r"|"
+            )
+        else:
+            test_regex = test_regex[:-1]
+
+        for item in self.test_conventions["class"]:
+            class_regex = (
+                class_regex + r"(?<=class\s).*{0}.*.[\w](?=\(|\:)".format(item) + r"|"
+            )
+        else:
+            class_regex = class_regex[:-1]
+        tests = []
+        last_file_name = None
+        file_dict = {}
+        class_name = ""
+        was_it_class = False
+        for candidate in raw_output.splitlines():
+            file_or_test = self.parser(candidate)
+            if file_or_test:
+                if re.search(test_file_regex, file_or_test):
+                    last_file_name = file_or_test
+                    if last_file_name not in file_dict.keys():
```

Comment:
Here you can use [setdefault](https://docs.python.org/3/library/stdtypes.html#dict.setdefault) and eliminate the if condition. This way, it will only set the default value if the key doesn't exist.

```suggestion
                    file_dict.setdefault(last_file_name, [])
```

A part from that, `was_it_class = False` doesn't depend on this condition, right?